File: lecture_code/house_robber.py
class Solution(object):
    def rob(self, nums):
        nums = [0, 0] + nums
        n = len(nums)
        dp = [0] * n
        for i in xrange(2, n):
            dp[i] = max(dp[i - 2] + nums[i], dp[i - 1])
        return dp[-1]

File: lecture_code/house_robber.java
public calss Solution {
    public int rob(int[], nums) {
	int n = nums.length;
	int[][] dp = new int[n+1][2];
	//dp[0][c] = 0
	for (int i = 1; i <= n; i++) {
	    dp[i][0] = Math.max(dp[i-1][1], dp[i-1][0]);
	    dp[i][1] = dp[i-1][0] + nums[i - 1];
	}
	return Math.max(dp[n][0], dp[n][1]);
    }
}

